src/serde/reader.rs

```diff
@@ -41,7 +41,11 @@ impl<R: Read, E: ByteOrder> Deserializer<R, E> {
     }
 
     fn read_bytes(&mut self, count: u64) -> Result<()> {
-        self.read += count;
+        if let (read, false) = self.read.overflowing_add(count) {
```

Comment:
Ah, good point, I see where this is happening.  Fortunately, I think there's a pretty easy solution:

Right now we are using read_bytes in two locations: 
* read_type
* read_vec

(I should probably rename one of those two methods, because despite the name, they are not similar at all).

`read_type` is safe, because it is only being used to increment the `read` counter for statically known types.
`read_vec` is not currently safe because it trusts the length that was read out of the byte-stream.  

This second case would be pretty easy to fix simply by sticking the calls to `read_bytes` inside of the `read_vec_small` and `read_vec_large` functions immediately after any calls to `read_exact`.